tools/itch5cacheaware.cpp

```diff
@@ -0,0 +1,147 @@
+#include <jb/itch5/compute_book_cache_aware.hpp>
+#include <jb/itch5/process_iostream.hpp>
+#include <jb/book_cache_aware_stats.hpp>
+#include <jb/fileio.hpp>
+#include <jb/log.hpp>
+
+#include <iostream>
+#include <stdexcept>
+#include <unordered_map>
+
+/**
+ * Compute ITCH5 Cache Aware statistics.
+ *
+ * Generate statistics per symbol and aggregated.
+ * See:
+ * https://github.com/GFariasR/jaybeams/wiki \
+ * /ITCH5-Cache-Aware-StatisticsProject
+ * for design and implementation details.
+ */
+namespace {
+
+class config : public jb::config_object {
+public:
+  config();
+  config_object_constructors(config);
+
+  void validate() const override;
+
+  jb::config_attribute<config, std::string> input_file;
+  jb::config_attribute<config, std::string> output_file;
+  jb::config_attribute<config, jb::log::config> log;
+  jb::config_attribute<config, jb::book_cache_aware_stats::config> stats;
+  jb::config_attribute<config, jb::book_cache_aware_stats::config> symbol_stats;
+  jb::config_attribute<config, bool> enable_symbol_stats;
+  jb::config_attribute<config, jb::itch5::tick_t> tick_offset;
+};
+
+} // anonymous namespace
+
+int main(int argc, char* argv[]) try {
+  config cfg;
+  cfg.load_overrides(
+      argc, argv, std::string("itch5cacheaware.yaml"), "JB_ROOT");
+  jb::log::init(cfg.log());
+
+  boost::iostreams::filtering_istream in;
+  jb::open_input_file(in, cfg.input_file());
+
+  boost::iostreams::filtering_ostream out;
+  jb::open_output_file(out, cfg.output_file());
+
+  // set order book offset value
+  jb::itch5::order_book_cache_aware::tick_offset(cfg.tick_offset());
+
+  std::map<jb::itch5::stock_t, jb::book_cache_aware_stats> per_symbol;
+  jb::book_cache_aware_stats stats(cfg.stats());
+
+  auto cb =
```

Comment:
Please consider using the new form for callbacks implemented in itch5inside, mold2inside, or itch5eventdepthstats.